presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestDetermineSemiJoinDistributionType.java

```diff
@@ -0,0 +1,312 @@
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        tester.close();
+        tester = null;
+    }
+
+    @Test
+    public void testRetainDistributionType()
+    {
+        assertDetermineSemiJoinDistributionType()
+                .on(p ->
+                        p.semiJoin(
+                                p.values(ImmutableList.of(p.symbol("A1")), ImmutableList.of(expressions("10"), expressions("11"))),
+                                p.values(ImmutableList.of(p.symbol("B1")), ImmutableList.of(expressions("50"), expressions("11"))),
+                                p.symbol("A1"),
+                                p.symbol("B1"),
+                                p.symbol("output"),
+                                Optional.empty(),
+                                Optional.empty(),
+                                Optional.of(REPLICATED)))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testPartitionWhenRequiredBySession()
+    {
+        int aRows = 10_000;
+        int bRows = 100;
+        assertDetermineSemiJoinDistributionType()
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, JoinDistributionType.PARTITIONED.name())
+                .overrideStats("valuesA", PlanNodeStatsEstimate.builder()
+                        .setOutputRowCount(aRows)
+                        .addSymbolStatistics(ImmutableMap.of(new Symbol("A1"), new SymbolStatsEstimate(0, 100, 0, 6400, 100)))
+                        .build())
+                .overrideStats("valuesB", PlanNodeStatsEstimate.builder()
+                        .setOutputRowCount(bRows)
+                        .addSymbolStatistics(ImmutableMap.of(new Symbol("B1"), new SymbolStatsEstimate(0, 100, 0, 640000, 100)))
+                        .build())
+                .on(p ->
+                        p.semiJoin(
+                                p.values(new PlanNodeId("valuesA"), aRows, p.symbol("A1", BIGINT)),
+                                p.values(new PlanNodeId("valuesB"), bRows, p.symbol("B1", BIGINT)),
+                                p.symbol("A1"),
+                                p.symbol("B1"),
+                                p.symbol("output"),
+                                Optional.empty(),
+                                Optional.empty(),
+                                Optional.empty()))
+                .matches(semiJoin(
+                        "A1",
+                        "B1",
+                        "output",
+                        Optional.of(PARTITIONED),
+                        values(ImmutableMap.of("A1", 0)),
+                        values(ImmutableMap.of("B1", 0))));
+    }
+
+    @Test
+    public void testReplicatesWhenRequiredBySession()
+    {
+        int aRows = 10_000;
+        int bRows = 10_000;
+        assertDetermineSemiJoinDistributionType()
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, JoinDistributionType.BROADCAST.name())
```

Comment:
Set broadcast limit here to something very small, just to make sure that when the `BROADCAST` is set explicitly, the size limit is ignored.